gui/scripts/syncthing/core/controllers/syncthingController.js

```diff
@@ -1230,17 +1230,19 @@ angular.module('syncthing.core')
                 .success(function (data) {
                     data.ignore = data.ignore || [];
 
-                    $('#editFolder').modal('hide');
-                    var textArea = $('#editIgnores textarea');
-
-                    textArea.val(data.ignore.join('\n'));
-
-                    $('#editIgnores').modal()
-                        .on('hidden.bs.modal', function () {
-                            $('#editFolder').modal();
-                        })
-                        .on('shown.bs.modal', function () {
-                            textArea.focus();
+                    $('#editFolder').modal('hide')
+                        .one('hidden.bs.modal', function() {
```

Comment:
That's deliberate. It's needed after `$('#editFolder').modal('hide')` because otherwise we register a handler which never gets removed, and so if we show the ignores dialog, then close it and try and close the folder dialog, the ignores dialog gets shown again.

In the other places, it's more semantically correct (in my understanding).